### PR TITLE
Add boundary for bezier curve

### PR DESCRIPTION
**Pull request by HaoxuanGuo**
Bezier curve for #210 .

### CODE REVIEW

File: src/primitives/bezier.jl [hunk 1]
```diff
@@ -101,3 +101,11 @@ function (curve::BezierCurve{Dim,T})(t, ::Horner) where {Dim,T}
   b₀ = bᵢ₋₁
   Point(b₀)
 end
+
+function boundary(curve::BezierCurve{Dim,T}) where {Dim,T}
+  return if ncontrols(curve) == 1
+    curve.controls[1]
+  else
+    PointSet(curve.controls[begin], curve.controls[end])
+  end
```
Comment (juliohm) [inline src/primitives/bezier.jl]: This is not following our code style. Also, we should probably return a `PointSet` always for type stability.

Comment (hx-guo) [reply src/primitives/bezier.jl]: @juliohm fixed.

File: src/primitives/bezier.jl [hunk 2]
```diff
@@ -101,3 +101,11 @@ function (curve::BezierCurve{Dim,T})(t, ::Horner) where {Dim,T}
   b₀ = bᵢ₋₁
   Point(b₀)
 end
+
+function boundary(curve::BezierCurve{Dim,T}) where {Dim,T}
+  return if ncontrols(curve) == 1
+    PointSet(curve.controls[1])
+  else
+    PointSet(curve.controls[begin], curve.controls[end])
+  end
```
Comment (juliohm) [inline src/primitives/bezier.jl]:
```suggestion
  points = curve.controls
  PointSet(unique([first(points), last(points)]))
```

File: src/primitives/bezier.jl [hunk 3]
```diff
@@ -101,3 +101,8 @@ function (curve::BezierCurve{Dim,T})(t, ::Horner) where {Dim,T}
   b₀ = bᵢ₋₁
   Point(b₀)
 end
+
+function boundary(curve::BezierCurve{Dim,T}) where {Dim,T}
```
Comment (juliohm) [inline src/primitives/bezier.jl]:
```suggestion
function boundary(curve::BezierCurve)
```

File: src/primitives/bezier.jl [hunk 4]
```diff
@@ -101,3 +101,8 @@ function (curve::BezierCurve{Dim,T})(t, ::Horner) where {Dim,T}
   b₀ = bᵢ₋₁
   Point(b₀)
 end
+
+function boundary(curve::BezierCurve)
+  points = curve.controls
+  PointSet(unique([first(points), last(points)]))
```
Comment (juliohm) [inline src/primitives/bezier.jl]:
```suggestion
  points = curve.controls
  a = first(points)
  b = last(points)
  a == b ? nothing : PointSet([a, b])
```

File: src/primitives/bezier.jl [hunk 5]
```diff
@@ -101,3 +101,10 @@ function (curve::BezierCurve{Dim,T})(t, ::Horner) where {Dim,T}
   b₀ = bᵢ₋₁
   Point(b₀)
 end
+
+function boundary(curve::BezierCurve)
+  points = curve.controls
+  a = first(points)
+  b = last(points)
```
Comment (juliohm) [inline src/primitives/bezier.jl]:
```suggestion
  p = curve.controls
  a = first(p)
  b = last(p)
```

File: test/primitives.jl [hunk 6]
```diff
@@ -49,6 +49,10 @@
       @test_throws DomainError(T(1.2), "b(t) is not defined for t outside [0, 1].") b(T(1.2), method)
     end
 
+    @test boundary(b) == PointSet(P2(0,0),P2(1,0))
+    b = BezierCurve(P2(0,0))
+    @test boundary(b) == PointSet(P2(0,0))
```
Comment (juliohm) [inline test/primitives.jl]:
```suggestion
    b = BezierCurve(P2(0,0), P2(1,1))
    @test boundary(b) == PointSet([P2(0,0), P2(1,1)])
```

File: test/primitives.jl [hunk 7]
```diff
@@ -49,6 +49,10 @@
       @test_throws DomainError(T(1.2), "b(t) is not defined for t outside [0, 1].") b(T(1.2), method)
     end
 
+    @test boundary(b) == PointSet(P2(0,0),P2(1,0))
```
Comment (juliohm) [inline test/primitives.jl]:
```suggestion
    @test boundary(b) == PointSet(P2(0,0), P2(1,0))
```

**Pull request merged**